File: backend/src/main/java/com/mysite/knitly/domain/payment/entity/PaymentMethod.java
package com.mysite.knitly.domain.payment.entity;

public enum PaymentMethod {
    // 카드, 가상계좌, 간편결제
    CARD, VIRTUAL_ACCOUNT, EASY_PAY;

    // 토스페이먼츠 API 응답의 method 값으로부터 enum 변환
    public static PaymentMethod fromString(String method) {
        if (method == null) {
            throw new IllegalArgumentException("결제수단 정보가 없습니다.");
        }

        return switch (method.toUpperCase()) {
            case "CARD", "카드" -> CARD;
            case "VIRTUALACCOUNT", "VIRTUAL_ACCOUNT", "가상계좌" -> VIRTUAL_ACCOUNT;
            case "EASYPAY", "EASY_PAY", "간편결제" -> EASY_PAY;
            default -> throw new IllegalArgumentException("지원하지 않는 결제수단입니다: " + method);
        };
    }
}

File: backend/src/main/java/com/mysite/knitly/domain/payment/entity/PaymentMethod.kt
package com.mysite.knitly.domain.payment.entity

enum class PaymentMethod {
    // 카드, 가상계좌, 간편결제, 무료 결제
    CARD, VIRTUAL_ACCOUNT, EASY_PAY, FREE;

    companion object {
        // 토스페이먼츠 API 응답의 method 값으로부터 enum 변환
        @JvmStatic
        fun fromString(method: String): PaymentMethod {
            requireNotNull(method) { "결제수단 정보가 없습니다." }

            return when (method.uppercase()) {
                "CARD", "카드" -> PaymentMethod.CARD
                "VIRTUALACCOUNT", "VIRTUAL_ACCOUNT", "가상계좌" -> PaymentMethod.VIRTUAL_ACCOUNT
                "EASYPAY", "EASY_PAY", "간편결제" -> PaymentMethod.EASY_PAY
                "FREE", "무료" -> PaymentMethod.FREE
                else -> throw IllegalArgumentException("지원하지 않는 결제수단입니다: " + method)
            }
        }
    }
}

